Description: bluetooth: set better priorities for profiles
 Since HSP had higher priority than A2DP, the default profile when
 connecting a new headset was HSP. To me it makes more sense to default
 to high-quality output. We already have some automatic policies to
 switch to HSP when it's needed.
 .
 I also made the A2DP source and HSP/HFP gateway profiles have lower
 priority than the A2DP sink and HSP headset profiles. The A2DP source
 and HSP/HFP gateway profiles should only be activated if the remote
 device initiates audio streaming, so it makes sense to have lower
 priority for those profiles.
Author: Tanu Kaskinen <tanuk@iki.fi>
Acked-by: Daniel van Vugt <daniel.van.vugt@canonical.com>
Origin: upstream, https://cgit.freedesktop.org/pulseaudio/pulseaudio/commit/src/modules?id=85daab2725c8964d5e3d07089c4056435022d12e
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1720684
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=103058
Last-Update: 2017-12-14

Index: pulseaudio/src/modules/bluetooth/module-bluez5-device.c
===================================================================
--- pulseaudio.orig/src/modules/bluetooth/module-bluez5-device.c
+++ pulseaudio/src/modules/bluetooth/module-bluez5-device.c
@@ -1851,7 +1851,7 @@ static pa_card_profile *create_card_prof
     switch (profile) {
     case PA_BLUETOOTH_PROFILE_A2DP_SINK:
         cp = pa_card_profile_new(name, _("High Fidelity Playback (A2DP Sink)"), sizeof(pa_bluetooth_profile_t));
-        cp->priority = 10;
+        cp->priority = 40;
         cp->n_sinks = 1;
         cp->n_sources = 0;
         cp->max_sink_channels = 2;
@@ -1863,7 +1863,7 @@ static pa_card_profile *create_card_prof
 
     case PA_BLUETOOTH_PROFILE_A2DP_SOURCE:
         cp = pa_card_profile_new(name, _("High Fidelity Capture (A2DP Source)"), sizeof(pa_bluetooth_profile_t));
-        cp->priority = 10;
+        cp->priority = 20;
         cp->n_sinks = 0;
         cp->n_sources = 1;
         cp->max_sink_channels = 0;
@@ -1875,7 +1875,7 @@ static pa_card_profile *create_card_prof
 
     case PA_BLUETOOTH_PROFILE_HEADSET_HEAD_UNIT:
         cp = pa_card_profile_new(name, _("Headset Head Unit (HSP/HFP)"), sizeof(pa_bluetooth_profile_t));
-        cp->priority = 20;
+        cp->priority = 30;
         cp->n_sinks = 1;
         cp->n_sources = 1;
         cp->max_sink_channels = 1;
@@ -1888,7 +1888,7 @@ static pa_card_profile *create_card_prof
 
     case PA_BLUETOOTH_PROFILE_HEADSET_AUDIO_GATEWAY:
         cp = pa_card_profile_new(name, _("Headset Audio Gateway (HSP/HFP)"), sizeof(pa_bluetooth_profile_t));
-        cp->priority = 20;
+        cp->priority = 10;
         cp->n_sinks = 1;
         cp->n_sources = 1;
         cp->max_sink_channels = 1;
